inst/include/dust/dust.hpp

```diff
@@ -235,6 +229,22 @@ class Dust {
   rng_t& pick_generator(const size_t i) {
     return _rng(i % _rng.size());
   }
+
+  void initialise(const init_t data, const size_t step,
+                  const size_t n_particles) {
+    _particles.clear();
+    _particles.reserve(n_particles);
+    for (size_t i = 0; i < n_particles; ++i) {
+      _particles.push_back(Particle<T>(data, step));
+    }
+
+    const size_t n = n_state_full();
+    _index.clear();
+    _index.reserve(n);
+    for (size_t i = 0; i < n; ++i) {
+      _index.push_back(i);
+    }
```

Comment:
Similar here, an overwrite in the loop would be simpler

Reply:
The index here *can* change size so we need to clear and push back so that the length is correct